samples/WebApi/README.md

```diff
@@ -0,0 +1,127 @@
+# Description
+This is a sample application, which uses CoreRT to compile a .NET Core Web API sample.
+TODO WIP - Might not work on Linix
+
+# Building a WebAPI app with CoreRT
+
+## Install the .NET Core SDK
+CoreRT is an AOT-optimized .NET Core runtime. If you're new to .NET Core make sure to visit the [official starting page](http://dotnet.github.io). It will guide you through installing pre-requisites and building your first app.
+If you're already familiar with .NET Core make sure you've [downloaded and installed the .NET Core 2 SDK](https://www.microsoft.com/net/download/core).
+
+## Create your app 
+
+Open a **new** shell/command prompt window and run the following commands.
+``` 
+> dotnet new webapi -o myApp 
+> cd myApp
+```
+
+## Add CoreRT to your project
+
+Using CoreRT to compile your application is done via the ILCompiler NuGet package.
+
+In your shell/command prompt navigate to the root directory of your project run the command 
+
+`> dotnet new nuget `
+
+This will add a nuget.config file to your application. Open the file and in the ``<packageSources> `` element under ``<clear/>`` add the following:
+
+```xml
+<add key="dotnet-core" value="https://dotnet.myget.org/F/dotnet-core/api/v3/index.json" /> 
+<add key="aspnet-core" value="https://dotnet.myget.org/F/aspnetcore-release/api/v3/index.json" /> 
+```
+
+Once you've added the package source, add a reference to the compiler by running the following command:
+
+`> dotnet add package Microsoft.DotNet.ILCompiler -v 1.0.0-alpha-* `
+
```

Comment:
I would add a new `##` chapter here. Changing Mvc to MvcCore is a topic on its own, and deserves more explanation on why it is necessary.

Reply:
How in-depth do we want to go? To keep the guide as lean as possible, I would go for 3/4 sentences on what's not supported yet. But more detail might be useful to someone who wants to use the sample as a starting point.

Reply:
It needs to be clear why it is done. It does not have to be too deep.

`AddMvc` has big and fat closure that you never want for full AOT (even if everything in it was supported). I think it can be something like this:

```
## Register required MVC services only

`AddMvc` call in the default template registers a large set of middleware services by default, even if they are not needed by the application. It is not friendly to AOT compilation because of it leads to large binaries, and it may hit unsupported features. Replace `AddMvc` with `AddMvcCore()` that adds the basic functionality only, followed by explicit registration of the services that are  required by the application:

...

See https://dzone.com/articles/the-difference-between-addmvc-and-addmvccore for more details about this.
```